### PR TITLE
Bluetooth: samples: Add ISO connected benchmark sample

### PR DESCRIPTION
**Pull request by Thalley**
Add a sample that measures packet loss in a connected
ISO setup.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/host/iso.c [hunk 1]
```diff
@@ -141,6 +141,8 @@ int hci_le_reject_cis(uint16_t handle, uint8_t reason)
 	struct net_buf *buf;
 	int err;
 
+	BT_ERR("reason %d", reason);
```
Comment (jhedberg) [inline subsys/bluetooth/host/iso.c:144]: Seems unrelated. Move to separate commit or just remove

File: subsys/bluetooth/host/iso.c [hunk 2]
```diff
@@ -236,6 +238,7 @@ void hci_le_cis_req(struct net_buf *buf)
 
 	err = hci_le_accept_cis(cis_handle);
 	if (err) {
+		BT_DBG("HCI could not accept ISO %d", err);
```
Comment (jhedberg) [inline subsys/bluetooth/host/iso.c:241]: Same here

**Review by jhedberg — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/iso.c (hunk 1 above)
Comment (Thalley) [reply subsys/bluetooth/host/iso.c]: Ah, yeah sorry, that was some debug statements I forgot to remove. Thanks!

**Review by jhedberg — APPROVED** [review]

File: samples/bluetooth/iso_connected_benchmark/src/main.c [hunk 3]
```diff
@@ -0,0 +1,1041 @@
...
+	.accept = iso_accept,
+};
+
+static bool data_cb(struct bt_data *data, void *user_data)
+{
+	char *name = user_data;
+	uint8_t len;
+
+	switch (data->type) {
+	case BT_DATA_NAME_SHORTENED:
```
Comment (koffes) [inline samples/bluetooth/iso_connected_benchmark/src/main.c:306]: Does the style guide mandate a /* Fall through */ comment here?

File: samples/bluetooth/iso_connected_benchmark/src/main.c [hunk 4]
```diff
@@ -0,0 +1,1041 @@
...
+	       qos->rtn, DEFAULT_CIS_RTN);
+
+	char_count = get_chars(buffer, sizeof(buffer) - 1);
+
+	if (char_count == 0) {
+		return DEFAULT_CIS_RTN;
+	}
+
+	rtn = strtoul(buffer, NULL, 0);
+	if (rtn > 16) {
```
Comment (koffes) [inline samples/bluetooth/iso_connected_benchmark/src/main.c:453]: This limit as well as the interval max/min, latency max, SDU interval, are they defined somewhere?

**Review by koffes — APPROVED** [review]

**Comment by ppryga** [comment]
Hi there! I'm wondering how this PR has passed CI when build of the sample fails in other PRs, e.g. https://buildkite.com/zephyr/zephyr/builds/34583.

Reason of failure is this option `CONFIG_BT_ISO=y` from prj.conf. The config entry is defined as hidded in `subsys\bluetooth\KConfig`.

**Comment by nashif** [comment]
this was reverted, please submit a new PR with a working sample.

File: samples/bluetooth/iso_connected_benchmark/sample.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+sample:
+  name: BLE ISO Connected
+  description: Bluetooth Low Energy ISO Connected Benchmark sample
+tests:
+  samples.bluetooth.iso_connected_benchmark:
+    build_only: true
+    build_on_all: true
```
Comment (nashif) [inline samples/bluetooth/iso_connected_benchmark/sample.yaml:7]: You should not be building on all, (build_on_all). Also, why build_only? Is this not supposed to run being a benchmark and a sample?
